pkg/vsphere/tasks/waiter.go

```diff
@@ -78,7 +85,44 @@ func WaitForResult(ctx context.Context, f func(context.Context) (ResultWaiter, e
 		}
 
 		log.Errorf(cerr.Error())
-		return nil, cerr
+		return info, cerr
 	}
 	return info, nil
 }
+
+func Retry(ctx context.Context, f func(context.Context) (ResultWaiter, error)) (*types.TaskInfo, error) {
+	r := rand.New(rand.NewSource(time.Now().UnixNano())) //creates a more unique random
+	var err error
+	var taskInfo *types.TaskInfo
+	backoffFactor := int64(1)
+
+	for {
+		taskInfo, err = WaitForResult(ctx, f)
+		if err != nil && taskInfo == nil {
```

Comment:
Just use Wait(ctx, f) instead of WaitForResult() - just seeing now that taskInfo isn't used for anything other than the error.  See: https://github.com/vmware/govmomi/blob/master/task/wait.go#L97-L111

If Wait() returns err != nil:
```go
if terr, ok:= err.(*task.Error); ok {
   if _, ok := terr.Fault().(types.TaskInProgressFault); ...
} else {
   return err // not a task.Error
}
```